File: problem1.go
// Time Complexity :  O(N)
// Space Complexity : O(1) 
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

func findDisappearedNumbers(nums []int) []int {
    if len(nums) == 0{
        return []int{}
    }
    
    result := []int{}
    
    for i:=0;i<len(nums);i++{
        index := int(math.Abs(float64(nums[i])))-1
        if nums[index] > 0{
            nums[index] = nums[index] * -1
        }
    }
    
    for key,val := range nums{
        if val > 0{
            result = append(result,key+1)
        }
    }
    
    return result
}

File: Problem1.cs
public class DisappearingNumbers
    {
        // Time Complexity : O(n)
        // Space Complexity : O(1) - since we are searching in place
        // Did this code successfully run on Leetcode : Yes
        // Any problem you faced while coding this : No
        public IList<int> FindDisappearedNumbers(int[] nums)
        {
            int n = nums.Length;
            IList<int> result = new List<int>();

            //loop through the array to check for existing numbers just like in hashset
            for(int i = 0; i < n; i++)
            {
                int index = Math.Abs(nums[i]) - 1;
                if(nums[index] > 0) //if the number is not negative already
                {
                    nums[index] = nums[index] * -1;
                }
            }

            //loop through the array again to check for missing numbers
            for(int i = 0; i < n; i++)
            {
                if(nums[i] < 0) //if the number is negative already then just make it positive to retain the original array values
                {
                    nums[i] = nums[i] * -1;
                }
                else
                {
                    result.Add(i + 1); // if number is positive then we have not found the index + 1 number anywhere in the array
                }
            }
            return result;
        }
    }
